#### Задача 1

- Скачать файл в уроке
- 
Считать данные с помощью panda s- 
Вывести на экран первые 5 стр о- к
Посмотреть на описание признаков и на их содержание

In [1]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('kc_house_data.csv', sep=',')
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

#### Задач- а 2

Проведите первичный анализ да- нных
Изучите типы д- анных
Найдите количество пропущенных ячеек в - данных
Посчитайте основные статистики по всем признакам и поизуч- айте их
Пишите выводы

In [7]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [9]:
df.describe(include='object')

date
count             21613
unique              372
top     20140623T000000
freq                142

In [14]:
df[['bedrooms', 'floors']].max()

bedrooms    33.0
floors       3.5
dtype: float64

In [18]:
df[['bedrooms', 'floors']].min()

bedrooms    0.0
floors      1.0
dtype: float64

In [17]:
df[['price', 'view', 'grade']].mean()

price    540088.141767
view          0.234303
grade         7.656873
dtype: float64

In [19]:
df[['price', 'view', 'grade']].max()

price    7700000.0
view           4.0
grade         13.0
dtype: float64

In [20]:
df[['price', 'view', 'grade']].min()

price    75000.0
view         0.0
grade        1.0
dtype: float64

In [21]:
df['waterfront'].unique()

array([0, 1], dtype=int64)

In [39]:
x = df.sort_values('price', ascending=False)
x

id             date      price  bedrooms  bathrooms  \
7252   6762700020  20141013T000000  7700000.0         6       8.00   
3914   9808700762  20140611T000000  7062500.0         5       4.50   
9254   9208900037  20140919T000000  6885000.0         6       7.75   
4411   2470100110  20140804T000000  5570000.0         5       5.75   
1448   8907500070  20150413T000000  5350000.0         5       5.00   
...           ...              ...        ...       ...        ...   
8274   3883800011  20141105T000000    82000.0         3       1.00   
16198  3028200080  20150324T000000    81000.0         2       1.00   
465    8658300340  20140523T000000    80000.0         1       0.75   
15293    40000362  20140506T000000    78000.0         2       1.00   
1149   3421079032  20150217T000000    75000.0         1       0.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
7252         12050     27600     2.5           0     3  ...     13   
3914         10040     37325     2.0           1     2  ...     11   
9254          9890     31374     2.0           0     4  ...     13   
4411          9200     35069     2.0           0     0  ...     13   
1448          8000     23985     2.0           0     4  ...     12   
...            ...       ...     ...         ...   ...  ...    ...   
8274           860     10426     1.0           0     0  ...      6   
16198          730      9975     1.0           0     0  ...      5   
465            430      5050     1.0           0     0  ...      4   
15293          780     16344     1.0           0     0  ...      5   
1149           670     43377     1.0           0     0  ...      3   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
7252         8570           3480      1910          1987    98102  47.6298   
3914         7680           2360      1940          2001    98004  47.6500   
9254         8860           1030      2001             0    98039  47.6305   
4411         6200           3000      2001             0    98039  47.6289   
1448         6720           1280      2009             0    98004  47.6232   
...           ...            ...       ...           ...      ...      ...   
8274          860              0      1954             0    98146  47.4987   
16198         730              0      1943             0    98168  47.4808   
465           430              0      1912             0    98014  47.6499   
15293         780              0      1942             0    98168  47.4739   
1149          670              0      1966             0    98022  47.2638   

          long  sqft_living15  sqft_lot15  
7252  -122.323           3940        8800  
3914  -122.214           3930       25449  
9254  -122.240           4540       42730  
4411  -122.233           3560       24345  
1448  -122.220           4600       21750  
...        ...            ...         ...  
8274  -122.341           1140       11250  
16198 -122.315            860        9000  
465   -121.909           1200        7500  
15293 -122.280           1700       10387  
1149  -121.906           1160       42882  

[21613 rows x 21 columns]

In [44]:
x = df[df['waterfront'] == 1][['price', 'bedrooms', 'waterfront', 'bathrooms']]
x.sort_values(['price', 'bedrooms'], ascending=False).head()

price  bedrooms  waterfront  bathrooms
3914  7062500.0         5           1       4.50
1315  5300000.0         6           1       6.00
1164  5110800.0         5           1       5.25
8092  4668000.0         5           1       6.75
2626  4500000.0         5           1       5.50

Выводы:

макс цена - 7700000
средняя цена - 540088
мин цена -  7500
макс кол-во спален - 33
мин кол-во спален - 0
макс кол-во этажей - 3,5
мин кол-во этажей - 1

#### Условие 3: 3 задача
 
Ответьте на несколько вопросо
в
3.1 В каком диапазоне изменяются стоимости недвижимости?
3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
3.3 Как много домов с разными этажами в данных?
3.4 Насколько хорошие состояния у домов в данных?
3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [46]:
df['price'].min(), df['price'].max()

(75000.0, 7700000.0)

In [54]:
df['sqft_living'].mean() * 100/ df['sqft_lot'].mean()

13.76781757959227

In [57]:
df['floors'].unique()

array([1. , 2. , 1.5, 3. , 2.5, 3.5])

In [60]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

In [67]:
df['condition'].value_counts()
# состояние дома

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

In [64]:
df['yr_built'].min(), df['yr_built'].max() 

(1900, 2015)

#### Условие 4: 4 задача
Ответьте на несколько вопросов

4.1 Сколько в среднем стоят дома, у которых 2 спальни?
4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
4.3 Как много домов коснулся ремонт?
4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [82]:
# 1
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

In [76]:
# 2
df[df['price'] > 600000]['sqft_living'].mean()

2895.47442455243

In [79]:
# 3
# yr_renovated
df[df['yr_renovated'] != 0].shape[0]

914

In [81]:
# 4
x = df[df['grade'] > 10]['price'].mean()
y = df[df['grade'] < 4]['price'].mean()
x - y

1488885.1175298805

#### Условие 5: 5 задача

Ответьте на несколько вопросов

5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [88]:
# 1
df[(df['bathrooms'] > 3) & (df['sqft_basement'] > 0)].shape[0]

1143

In [93]:
# 2
x = df[
((df['view'] == 4) |(df['waterfront'] == 1)) & 
(df['condition'] > 4) & 
(df['yr_built'] > 1980)
]
x['price'].min(), x['price'].max()

(1295000.0, 3000000.0)

In [92]:
# 3
x = df[
(df['sqft_basement'] == 0) &
(df['floors'] == 2) &
(df['price'] < 150000)
]
x['condition'].mean()

2.8333333333333335